rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/are_all_terminal_heating_loops_attached_to_boiler.py

```diff
@@ -0,0 +1,58 @@
+from rct229.data.schema_enums import schema_enums
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_all, find_exactly_one_with_field_value
+
+FLUID_LOOP_TYPE = schema_enums["FluidLoopOptions"]
+
+
+def are_all_terminal_heating_loops_attached_to_boiler(rmi_b, terminal_unit_id_list):
+    """Returns TRUE if the fluid loop associated with the heating_from_loop associated with each terminal unit is attached to a boiler. Returns FALSE if this is not the case.
+
+    Parameters
+    ----------
+    rmi_b : json
+        RMD at RuleSetModelInstance level
+    terminal_unit_id_list : list
+        List of terminal units IDs
+
+    Returns
+    -------
+    bool
+        True: fluid loop associated with the heating_from_loop associated with each terminal unit is attached to a boiler
+        False: otherwise
+    """
+    are_all_terminal_heating_loops_attached_to_boiler_flag = True
+    boilers = find_all("$.boilers[*]", rmi_b)
+    loop_boiler_dict = dict()
+    for boiler_b in boilers:
+        loop_id = getattr_(boiler_b, "boiler", "loop")
```

Comment:
Could you check if ```boiler``` is meant to be ```boilers```?

Reply:
Which `boiler` are you referring to?

Reply:
@yunjoonjung-PNNL I suspect you are referring to the "boiler" as the second argument on line 28.  That argument is intended to be a string representing the type of entity the first argument represents. That string appears in the exception that is raised if there is no loop in boiler_b.